### PR TITLE
LogAppender for java platform/system logger

### PR DESCRIPTION
**Pull request by justcoon**
similar to `slf4j` backend implementation

default behavior:
- all annotations are placed at the beginning of log message
- cause - LogAppender.appendCause - as throwable, Cause transformed into FiberFailure
- logger name extracted from `Trace`, added `JPL.loggerName` zio LogAnnotation for custom logger name
- updated documentation and examples

open questions:
- `slf4j` and this `jpl` implementation got custom logger name aspect, should not we do common one?
-  by default both are  appending all LogAnnotations-s, including logger name annotation, there is question if this annotation should not be excluded in default setup, because if logger name is extracted from Trace, there is not logger name annotation

related to  #322 (from this ticket it is not clear if this was requirement for backend implementation like slf4j or it should be something similar to slf4j-bridge)

### CODE REVIEW

**Comment by justcoon** [comment]
NOTE: 
CI setup needs to be updated, as java platform/system logger was introduced at [`JDK9`](https://openjdk.org/jeps/264)

File: jpl/src/main/scala/zio/logging/backend/JPL.scala [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+  val logFormatDefault: LogFormat = LogFormat.allAnnotations + LogFormat.line + LogFormat.cause
+
+  def loggerName(value: String): ZIOAspect[Nothing, Any, Nothing, Any, Nothing, Any] =
+    new ZIOAspect[Nothing, Any, Nothing, Any, Nothing, Any] {
+      def apply[R, E, A](zio: ZIO[R, E, A])(implicit trace: Trace): ZIO[R, E, A] =
+        ZIO.logAnnotate(loggerNameKey, value)(zio)
+    }
+
+  def getLoggerName(default: String = "zio-jpl-logger"): Trace => String =
```
Comment (jdegoes) [inline jpl/src/main/scala/zio/logging/backend/JPL.scala:43]: This should be hidden / made private.

File: jpl/src/main/scala/zio/logging/backend/JPL.scala [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    loggerName: Trace => String
+  ): ZLayer[Any, Nothing, Unit] =
+    Runtime.addLogger(jplLogger(format, loggerName))
+
+  def jpl(
+    format: LogFormat
+  ): ZLayer[Any, Nothing, Unit] =
+    jpl(format, getLoggerName())
+
+  def jpl: ZLayer[Any, Nothing, Unit] =
```
Comment (jdegoes) [inline jpl/src/main/scala/zio/logging/backend/JPL.scala:112]: This can be a `val`.

**Review by jdegoes — CHANGES_REQUESTED** [review]
Just minor tweaks, and then good to merge!

**Comment by justcoon** [comment]
@jdegoes - comments solved

**Review by jdegoes — APPROVED** [review]

**Pull request merged**